src/flutter/shell/platform/linux_embedded/window/native_window_drm_eglstream.cc

```diff
@@ -0,0 +1,299 @@
+    return DismissCursor();
+  }
+
+  // todo: implement here.
+  return true;
+}
+
+bool NativeWindowDrmEglstream::DismissCursor() {
+  // todo: implement here.
+  return true;
+}
+
+std::unique_ptr<SurfaceGlDrm<uint32_t, ContextEglDrmEglstream>>
+NativeWindowDrmEglstream::CreateRenderSurface() {
+  return std::make_unique<SurfaceGlDrm<uint32_t, ContextEglDrmEglstream>>(
+      std::make_unique<ContextEglDrmEglstream>(
+          std::make_unique<EnvironmentEglDrmEglstream>()));
+}
+
+bool NativeWindowDrmEglstream::ConfigureDisplayForEglstream() {
+  if (SetDrmClientCapability() != 0) {
+    LINUXES_LOG(ERROR) << "Couldn't set drm client capability";
+    return false;
+  }
+
+  auto plane_resources = drmModeGetPlaneResources(drm_device_);
+  if (!plane_resources) {
+    LINUXES_LOG(ERROR) << "Couldn't get plane resources";
+    return false;
+  }
+  drm_plane_id_ = FindPlane(plane_resources);
+  drmModeFreePlaneResources(plane_resources);
+  if (drm_plane_id_ == -1) {
+    LINUXES_LOG(ERROR) << "Could not find a plane.";
+    return false;
+  }
+
+  auto atomic = drmModeAtomicAlloc();
+  if (!atomic) {
+    LINUXES_LOG(ERROR) << "Couldn't allocate atomic";
+    return false;
+  }
+  int ret = -1;
+  if (AssignAtomicRequest(atomic)) {
+    ret = drmModeAtomicCommit(drm_device_, atomic,
+                              DRM_MODE_ATOMIC_ALLOW_MODESET, nullptr);
+  }
+  drmModeAtomicFree(atomic);
+  if (ret != 0) {
+    LINUXES_LOG(ERROR) << "Failed to commit an atomic property change request";
+    return false;
+  }
+
+  return true;
+}
+
+int NativeWindowDrmEglstream::SetDrmClientCapability() {
+  if (drmSetClientCap(drm_device_, DRM_CLIENT_CAP_UNIVERSAL_PLANES, 1) == 0) {
+    return drmSetClientCap(drm_device_, DRM_CLIENT_CAP_ATOMIC, 1);
+  }
+  return -1;
+}
+
+uint32_t NativeWindowDrmEglstream::FindPlane(drmModePlaneResPtr resources) {
+  for (uint32_t i = 0; i < resources->count_planes; i++) {
+    auto plane = drmModeGetPlane(drm_device_, resources->planes[i]);
+    if (plane) {
+      drmModeFreePlane(plane);
+
+      constexpr char kPropNamePlaneType[] = "type";
+      auto type = GetPropertyValue(resources->planes[i], DRM_MODE_OBJECT_PLANE,
+                                   kPropNamePlaneType);
+      if (type == DRM_PLANE_TYPE_PRIMARY) {
+        return resources->planes[i];
+      }
+    }
+  }
+  // no plane found
+  return -1;
+}
+
+uint64_t NativeWindowDrmEglstream::GetPropertyValue(uint32_t id, uint32_t type,
+                                                    const char* prop_name) {
+  uint64_t value = -1;
+  auto properties = drmModeObjectGetProperties(drm_device_, id, type);
+  if (properties) {
+    for (uint32_t i = 0; i < properties->count_props; i++) {
+      auto property = drmModeGetProperty(drm_device_, properties->props[i]);
+      if (property) {
+        if (strcmp(prop_name, property->name) == 0) {
+          value = properties->prop_values[i];
+          drmModeFreeProperty(property);
+          break;
```

Comment:
Doesn't it need `drmModeFreeObjectProperties(properties);` ?

Reply:
`drmModeFreeObjectProperties(properties);` is executed after breaking for-loop.